kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -2,18 +2,85 @@ use ark_ff::{batch_inversion_and_mul, FftField};
 use ark_poly::{EvaluationDomain, Evaluations, Radix2EvaluationDomain as D};
 use rayon::prelude::*;
 
-/// The evaluations of all normalized lagrange basis polynomials at a given
-/// point. Can be used to evaluate an `Evaluations` form polynomial at that point.
+/// Evaluations of all normalized lagrange basis polynomials at a given point.
+/// Can be used to evaluate an `Evaluations` form polynomial at that point.
+///
+/// The Lagrange basis for polynomials of degree <= d over a domain
+/// {ω_0,...,ω_d} is the set of (d+1) polynomials ${l_0,...,l_d}$ of degree d
+/// that equal 1 at ω_i and 0 in the rest of the domain terms. They can be used
+/// to evaluate polynomials in evaluation form efficiently in O(d) time.
+///
+/// When chunking is in place, the domain size is larger than the maximum
+/// polynomial degree allowed. In particular, the SRS is not long enough to
+/// support higher degrees. Thus, we cannot obtain a polynomial f with degree
+/// c·n (with domain size n and number of chunks c) as usual with the equation:
+///
+/// f(X) = x_0 · l_0(X) + ... + x_n · l_n(X) + ... + x_{c*n} · l_{c*n}(X)
+///
+/// Instead, this struct will contain the (c*n+1) coefficients of the polynomial
+/// that is equal to the powers of the point x in the positions corresponding to
+/// the chunk, and 0 elsewhere in the domain. This is useful to evaluate the
+/// chunks of polynomials of degree c·n given in evaluation form at the point.
+///
 pub struct LagrangeBasisEvaluations<F> {
+    /// If no chunking:
+    /// - evals is a vector of length 1 containing a vector of size n
+    ///   corresponding to the evaluations of the Lagrange polynomials, which
+    ///   are the polynomials that equal 1 at ω_i and 0 elsewhere in the domain.
+    /// If chunking (a vector of size c * N )
+    /// - the first index refers to the chunks
+    /// - the second index refers j-th coefficient of the i-th chunk of the
+    ///   polynomial that equals the powers of the point and 0 elsewhere.
+    ///
     evals: Vec<Vec<F>>,
 }
 
 impl<F: FftField> LagrangeBasisEvaluations<F> {
-    /// Given the evaluations form of a polynomial, directly evaluate that polynomial at a point.
+    /// Given the evaluations form of a polynomial, directly evaluate that
+    /// polynomial at a point.
+    ///
+    /// The Lagrange basis evaluations can be used to evaluate a polynomial of
+    /// given in evaluation form efficiently in O(n) time, of degree d <= n
+    /// where n is the domain size, without the need of interpolating it first.
+    ///
+    /// Recall that a polynomial can be represented as the sum of the scaled
+    /// Lagrange polynomials using its evaluations on the domain:
+    /// $ f(x) = x_0 · l_0(x) + ... + x_n · l_n(x) $
```

Comment:
```suggestion
    /// `f(x) = x_0 · l_0(x) + ... + x_n · l_n(x)`
```

It is markdown, not LaTeX.